manifold/src/impl.cpp

```diff
@@ -382,9 +406,14 @@ int Manifold::Impl::InitializeNewReference(
     const std::vector<float>& properties,
     const std::vector<float>& propertyTolerance) {
   meshRelation_.triBary.resize(NumTri());
+
+  meshID2OriginalMutex_.lock();
   const int nextMeshID = meshID2Original_.size();
   meshID2Original_.push_back(nextMeshID);
+  meshID2OriginalMutex_.unlock();
```

Comment:
Could we make this locking simpler by making `meshID2Original` a hash table or something? The IDs don't have to be sequential; I'd be fine with a GUID if that would make the code simpler.

Reply:
I think hash table in the stl do not support concurrent access.